solutions/usaco-831.mdx

```diff
@@ -3,14 +3,116 @@ id: usaco-831
 source: USACO Bronze 2018 Open
 title: Team Tic Tac Toe
 author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma
+contributor: Aadit Ambadkar
 ---
 
 ## Solution 1
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_tttt_bronze_open18.html)
 
 <LanguageSection>
+<CPPSection>
 	
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+char board[3][3];
+
+// Does 1 cow win?
+int cowWins(char ch) {
+	// Check diagonals
+	if (board[0][0] == ch && board[1][1] == ch && board[2][2] == ch) {
+		return 1;
+	}
+	if (board[0][2] == ch && board[1][1] == ch && board[2][0] == ch) {
+		return 1;
+	}
+
+	// Check rows and columns
+	for (int i = 0; i < 3; i++) {
+		if (board[0][i] == ch && board[1][i] == ch && board[2][i] == ch) {
+			return 1;
+		}
+		if (board[i][0] == ch && board[i][1] == ch && board[i][2] == ch) {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+// Test if a team wins based on 3 characters in a row, column, or diagonal
+bool check3(char ch1, char ch2, char a, char b, char c) {
+	// All 3 characters have to be either ch1 or ch2
+	if (a != ch1 && a != ch2) {
+		return false;
+	}
+	if (b != ch1 && b != ch2) {
+		return false;
+	}
+	if (c != ch1 && c != ch2) {
+		return false;
+	}
+
+	// ch1 and ch2 have to appear at least once each
+	if (a != ch1 && b != ch1 && c != ch1) {
+		return false;
+	}
+	if (a != ch2 && b != ch2 && c != ch2) {
+		return false;
+	}
+
+	return true;
+}
+// Does a team win?
+int teamWins(char ch1, char ch2) {
```

Comment:
c++ uses snake_case, best to be consistent

Reply:
ok

Reply:
fwiw [google's style guide](https://google.github.io/styleguide/cppguide.html#Function_Names) begs to differ